### DISCUSSION TITLE
Odd behaviour of performance graph

### DISCUSSION THREAD

**Discussion by Neurocranium** [ghostfolio/ghostfolio#1671] [Q&A]
Hi community,

I just started my journey into ghostfolio and encountered a weird thing with the performance graphs I'd like to understand better.

As a starting point I just entered a single fund I own since 2016 into ghostfolio. My first purchase of the fund was on 22/02/2016 and I use it as a deposit for my landlord. So, aside from a few accumulation purchases and a few sells to cover for broker fees nothing is happening with this particular fund.

In the performance graph, there is a -23% drop in performance of my portfolio straight away after the initial purchase I don't get at all.
![image](https://user-images.githubusercontent.com/92733021/217322192-f91f10af-7220-4e25-a513-9e713002e008.png)

I already checked the historical data for the fund (https://finance.yahoo.com/quote/DZ7X.F/) and there certainly wasn't a drop in market price I could use to explain away this performance drop. Even with considering the initial issuing surcharge of the fund (19.70€ issuing price vs. 19.21€ market price on 29/02/2016) I can't amount it up to 23% performance loss.

Do I miss something here? Any suggestions how ghostfolio might come up with this performance loss? Or is there maybe some bug involved.

Best regards,
Sebastian

**Comment by dtslvr** [comment]
Hi @Neurocranium

I have quickly checked your symbol, but I could not find any [historical data](https://finance.yahoo.com/quote/DZ7X.F/history?period1=1451606400&period2=1675728000&interval=1d&filter=history&frequency=1d&includeAdjustedClose=true) for `DZ7X.F`. Is it perhaps related to that?

Have you also tracked fees? What does the chart look like for just this holding?

**Reply by Neurocranium** [reply]
Oh, true, the historical data I had in mind came from another web page and the internal ghostfolio query to Yahoo is indeed throwing an error.
![image](https://user-images.githubusercontent.com/92733021/217386012-0f939de6-26f7-4009-bab8-f2964f7119c8.png)

Is there maybe a way to change the data provider? I saw that there is also a connector for Alpha Vantage for example.

Maybe my way of modelling the account fees and the purchases from accumulation, might also throw the performance off.
![image](https://user-images.githubusercontent.com/92733021/217386887-6c2bcb89-5970-43ae-81ea-dc2cb40c57e5.png)

The fees i modeled by doing a sell for a price of 0€ to reflect the decrease in shares I own and the broker sold to cover the fee. Then I just added the 8€ of broker fee as a kind of "virtual" fee to this sell transaction.

For the shares I got via accumulation I first added a dividend for them and then on the same day a purchase in the size of the dividend without any fees.

Does this way of modelling make sense? Or do I throw some calculation off without realizing?

**Reply by Neurocranium** [reply]
Hi @dtslvr ,

in the meantime I added the historical data for the symbol to my database. Unfortunately, this doesn't change a thing on the performance graph. Currently I also have only this holding in my setup. Want to figure out something like this first before I put the rest of my portfolio in.

The performance graph and also the portfolio evaluation still have this sudden drop after my initial purchase:
![image](https://user-images.githubusercontent.com/92733021/218322506-9e13971b-25c6-43eb-a376-56215ce20139.png)
 ![image](https://user-images.githubusercontent.com/92733021/218322163-b60c3e84-faa1-4750-89fc-6a5c39297a97.png)
vs
![image](https://user-images.githubusercontent.com/92733021/218322182-91505115-f29a-4600-a9c8-de77bed90c36.png)

This is like a ~317€ drop out of nowhere. It certainly is not related to the price of the fund. That even went up for a while after my initial purchase.
![image](https://user-images.githubusercontent.com/92733021/218322272-dbfae7f7-5a4b-4be2-b8f0-3e2d53f99896.png)

I also changed my initial purchase transaction from 19,70€ price (including the issuance fee) to 19,12€ (market prices) + 45,63€ fee to better reflect the price vs. fee part. But both variants of modelling have this big drop.
![image](https://user-images.githubusercontent.com/92733021/218322636-e346b737-ea6d-41f8-a47f-10d7cdb02610.png)

Any further suggestions where to look?

**Reply by Neurocranium** [reply]
Well, not a solution but I found a good enough work around. I just moved the date of my initial purchase one week and one day into the future (from 22.02.2016 to 01.03.2016) and the whole graphs magically are fine then. ¯\\_ (ツ)_/¯

![image](https://user-images.githubusercontent.com/92733021/219802248-5a3c5e9e-1ac6-44cb-9e18-2e0fa525a54d.png)

Maybe, just as a thought: By doing this I moved the initial purchase to after the leap day we had in 2016. Perhaps, this day is somehow throwing the graphs off track. At least with having the purchase on the 29.02.2016 the graph was still off.
